doc/source/development_and_release_processes.rst

```diff
@@ -0,0 +1,238 @@
+* |chkbx| *Idiomatic packages* depend on stable releases of *auto-generated packages* only.
+* |chkbx| Release candidate *idiomatic packages* are available on PyPI.
+* |chkbx| Internal packages with runtime dependencies on *idiomatic packages* depend on the release candidate PyPI
+  releases.
+
+Phase 3: Pre-release
+~~~~~~~~~~~~~~~~~~~~
+
+#. Update the *production test VM* to run the Granta MI release validated during the hardening phase.
+#. Update the latest *idiomatic package* ``release`` branches created during the hardening phase to run CI against the
+   *production test VM*.
+#. Update the *PyGranta meta-package* ``main`` branch to depend on the public *idiomatic packages* release candidates
+   and *auto-generated packages* stable releases.
+#. Create a ``release`` branch for the *PyGranta meta-package* and publish as a release candidate to public PyPI.
+#. Create a Pull Request in the PyAnsys meta-package repository to move the PyGranta dependency to the release candidate
+   published in the preceding step.
+#. Manually test the *PyGranta meta-package* release candidate against the Granta MI release candidate.
+
+Testing should occur up to a week before the availability of Granta MI to customers. Any issues raised during testing
+should be immediately triaged and fixed or deferred.
+
+Checklist for completing this phase:
+
+* |chkbx| The *production test VM* is running the release version of Granta MI.
+* |chkbx| CI for all *idiomatic packages* is passing.
+* |chkbx| Release candidate *PyGranta meta-package* is available on PyPI.
+* |chkbx| The PyAnsys meta-package depends on the *PyGranta meta-package* release candidate.
+* |chkbx| Manual testing has been completed against the *PyGranta meta-package* and all *idiomatic packages*.
+* |chkbx| The GitHub PyGranta Development project contains no must-deliver features or bug fixes still outstanding for
+  the active release.
+
+Phase 4: Release
+~~~~~~~~~~~~~~~~
+
+The PyAnsys meta-package is typically released a week before Granta MI is available to customers. As a result, the
+stable versions of the packages should be released at least the week before Granta MI is available to customers:
+
+#. Defer any issues still associated with the active release in the GitHub PyGranta Development to a future release.
+#. Publish *idiomatic packages* as stable releases to public PyPI.
+#. Update the *PyGranta meta-package* ``main`` branch to depend on the public *idiomatic packages* stable releases.
+#. Cherry-pick the changes to the ``release`` branch.
+#. Publish the *PyGranta meta-package* as a stable release to public PyPI.
+#. Create a Pull Request in the PyAnsys meta-package repository to move the *PyGranta meta-package* dependency to the
+   new stable release.
+#. If any new *idiomatic package* releases are compatible with the previous release of Granta MI, create a patch release
+   of the previous *PyGranta meta-package* release.
```

Comment:
We can't do this if an older Python version has been dropped, because that version of PyGranta package will not be able to build with the rest of the PyAnsys packages.

If there's a specific issue that we're trying to resolve and a Python version has just been dropped, we should backport that release to the previous idiomatic package version as a patch release, and upgrade the PyGranta metapackage to *that* version.